### PR TITLE
go compiler check descriptive error message

### PR DESCRIPTION
**Pull request by simonleary-umass-edu**
This is a continuation of [my last PR](https://github.com/apptainer/apptainer/pull/1045) which I broke while fumbling around with `git`.

### CODE REVIEW

File: mlocal/checks/basechecks.chk [hunk 1]
```diff
@@ -62,13 +62,11 @@ fi
...
-			gopth=`command -v $go`
-			if [ "$gopth" != "" ]; then
-				hstgo=$gopth
-				break
-			fi
-		done
+		gopth=`command -v $go`
+		if [ "$gopth" != "" ]; then
+			hstgo=$gopth
+			break
```
Comment (DrDaveD) [inline mlocal/checks/basechecks.chk:68]: The `break` is no longer needed now that there's no loop.

File: mlocal/checks/basechecks.chk [hunk 2]
```diff
@@ -79,22 +77,19 @@ if [ $do_go_version_check = 0 ]; then
...
 	printf " checking: host Go compiler (at least version $hstgo_version)... "
-	for go in $hstgo_opts; do
-		if $makeit_scriptsdir/check-min-go-version $go $hstgo_version >/dev/null 2>&1; then
-			hstgo=`command -v $go`
-			break
-		fi
-	done
+	if $makeit_scriptsdir/check-min-go-version go $hstgo_version; then
+		hstgo=`command -v $go`
+		break
```
Comment (DrDaveD) [inline mlocal/checks/basechecks.chk:82]: likewise here.

File: mlocal/scripts/check-min-go-version [hunk 3]
```diff
@@ -27,19 +30,26 @@ MINVERSION="$2"
 MINMAJOR="${MINVERSION/.*}"
 MINMINOR="${MINVERSION#*.}"
 MINPATCH="${MINMINOR#*.}"
+
+insufficient_go_version(){
+    echo "found go $GOVERSION does not meet minimum requirement $MINVERSION!"
+}
```
Comment (DrDaveD) [inline mlocal/scripts/check-min-go-version:36]: Don't forget to pull the `exit 1` into this function.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Comment by simonleary-umass-edu** [comment]
Oops, I had fixed `exit 1` before but it got left behind when I was fumbling with git

File: mlocal/checks/basechecks.chk [hunk 4]
```diff
@@ -62,13 +62,10 @@ fi
...
 	if [ "$hostgo" = "" ]; then
 		printf " checking: looking for Go compiler... "
-		for go in $hstgo_opts; do
-			gopth=`command -v $go`
-			if [ "$gopth" != "" ]; then
-				hstgo=$gopth
-				break
-			fi
-		done
+		gopth=`command -v $go`
```
Comment (DrDaveD) [inline mlocal/checks/basechecks.chk:65]: This needs to be `go`, not `$go`.

File: mlocal/checks/basechecks.chk [hunk 5]
```diff
@@ -79,22 +76,18 @@ if [ $do_go_version_check = 0 ]; then
...
 elif [ "$hstgo" = "" ]; then
 	printf " checking: host Go compiler (at least version $hstgo_version)... "
-	for go in $hstgo_opts; do
-		if $makeit_scriptsdir/check-min-go-version $go $hstgo_version >/dev/null 2>&1; then
-			hstgo=`command -v $go`
-			break
-		fi
-	done
+	if $makeit_scriptsdir/check-min-go-version go $hstgo_version; then
+		hstgo=`command -v $go`
```
Comment (DrDaveD) [inline mlocal/checks/basechecks.chk:80]: Here too.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Review by DrDaveD — APPROVED** [review]

**Pull request merged**